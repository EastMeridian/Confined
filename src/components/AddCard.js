import React from 'react';
import t from 'prop-types';
import colors from 'styles/colors';
import { cardLayout, fullSizeLayout } from 'styles/layouts';
import ButtonBase from '@material-ui/core/ButtonBase';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import View from './View';

const containerStyle = {
  ...cardLayout,
  border: `1px dashed ${colors.secondaryText}`,
  alignItems: 'center',
  justifyContent: 'center',
};

const buttonStyle = fullSizeLayout;

const AddCard = ({ onClick }) => (
  <View style={containerStyle}>
    <ButtonBase style={buttonStyle} onClick={onClick}>
      <AddCircleOutlineIcon fontSize="large" style={{ color: colors.primary }} />
    </ButtonBase>
  </View>
);

AddCard.propTypes = {
  onClick: t.func,
};

AddCard.defaultProps = {
  onClick: () => {},
};

export default AddCard;
