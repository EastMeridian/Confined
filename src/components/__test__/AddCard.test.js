import '@testing-library/jest-dom';
import { create } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import AddCard from '../AddCard';

describe('AddCArd component', () => {
  it('should matches the snapshot', () => {
    const addCard = create(<AddCard />);
    expect(addCard.toJSON()).toMatchSnapshot();
  });

  it('should be clickable', () => {
    let count = 0;
    const { getByRole } = render(<AddCard onClick={() => { count += 1; }} />);
    fireEvent.click(getByRole('button'));
    expect(count).toEqual(1);
  });
});
