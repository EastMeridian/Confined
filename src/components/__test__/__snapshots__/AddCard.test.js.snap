// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCArd component should matches the snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "border": "1px dashed #9E9E9B",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "column",
      "height": "12em",
      "justifyContent": "center",
      "maxWidth": "82vw",
      "overflow": "hidden",
      "position": "relative",
      "width": "20em",
    }
  }
>
  <button
    className="MuiButtonBase-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      role="presentation"
      style={
        Object {
          "color": "#FD1708",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
  </button>
</div>
`;
